### PR TITLE
Feature/view profile page

### PR DESCRIPTION
**Pull request by eugeneiv1**
## Describe your changes

- add possibility to view public profile page of vendor/buyer
- add components of reviews and products
- add userApiSlice to fetch user data from server
- add follow feature

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://olga-team.atlassian.net/browse/TP-181?atlOrigin=eyJpIjoiNWZlMjA3MmM4NTNiNGM5Njg2ZDE0Yzk5YWEwMGQ1MTMiLCJwIjoiaiJ9)

followinf_buyer
Products
Reviews

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit f

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/productCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import {
+  ProductImageWrapper,
+  ProductInfo,
+  ProductName,
+  ProductPrice,
+  VendorName,
+} from 'components/productCard/styles.ts';
+import Carousel from 'components/carousel';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (MishaFomenko) [inline src/components/productCard/index.tsx:11]: commented code

File: src/components/productCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const ProductCard = ({ imageUrl }) => {
+  return (
+    <li>
+      <ProductImageWrapper>
+        <Carousel />
+      </ProductImageWrapper>
+      <ProductName>Product Name</ProductName> {/*TODO change to fetched data*/}
+      <ProductInfo>
+        <div>
+          <ProductPrice>$ 213,99</ProductPrice>
```
Comment (MishaFomenko) [inline src/components/productCard/index.tsx:21]: price constant

File: src/components/publicProfileInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { PublicProfileInfoType } from 'components/publicProfileInfo/types.ts';
+
+const PublicProfileInfo = ({
+  userRole,
+  userReviews,
+  userProducts,
+}: PublicProfileInfoType) => {
+  return (
+    <PublicProfileInfoWrapper>
+      {userRole === 'buyer' ? (
```
Comment (MishaFomenko) [inline src/components/publicProfileInfo/index.tsx]: we should not compare with strings

File: src/components/publicProfileInfo/styles.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    color: ${({ theme }) => theme.colors.primary_text};
+    background-color: ${({ theme }) => theme.colors.greyish_red};
+
+    cursor: pointer;
+  }
+
+  .react-tabs__tab--selected {
+    width: 50%;
+    padding: 10px 0;
+    background-color: #333333;
```
Comment (MishaFomenko) [inline src/components/publicProfileInfo/styles.ts]: const

File: src/components/publicProfileInfo/types.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { ResponseGetProduct, UserReview } from '../../redux/types';
```
Comment (MishaFomenko) [inline src/components/publicProfileInfo/types.ts]: absolute path

File: src/components/publicProfileSidebar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        setError(errMsg);
+      } else if (isErrorWithMessage(err)) {
+        setError(err.message);
+      }
+    }
+  };
+
+  return (
+    <SideBarWrapper>
+      {/*TEXT IN ELEMENTS IS NOT FINAL RESULT!!! Just during feature in progress*/}
```
Comment (MishaFomenko) [inline src/components/publicProfileSidebar/index.tsx:54]: ok

File: src/components/publicProfileSidebar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  const { data: isFollowing, refetch } = useGetIsFollowingQuery(userId!);
+  const [followUser] = useFollowUserMutation();
+
+  const handleFollow = async () => {
+    try {
+      await followUser(userId!);
+      refetch();
+    } catch (err) {
+      if (isFetchBaseQueryError(err)) {
+        const errMsg = 'error' in err ? err.error : JSON.stringify(err.data);
```
Comment (MishaFomenko) [inline src/components/publicProfileSidebar/index.tsx:44]: comparing with strings

File: src/components/publicProfileSidebar/styles.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  width: 100%;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  margin-top: 32px;
+  padding: 14px 0;
+
+  border: 1px solid #edeae9;
```
Comment (MishaFomenko) [inline src/components/publicProfileSidebar/styles.ts]: color const

File: src/components/publicProfileSidebar/styles.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  border-radius: 12px;
+`;
+
+export const FollowButton = styled.button<{ theme?: Theme }>`
+  width: 61%;
+
+  font-family: ${({ theme }) => theme.fontNames.playfairDisplay};
+  text-align: center;
+  color: ${({ theme }) => theme.colors.white};
+  font-weight: 500;
```
Comment (MishaFomenko) [inline src/components/publicProfileSidebar/styles.ts]: const

File: src/components/publicProfileInfo/styles.ts (hunk 4 above)
Comment (AlexandraVoloshyna) [inline src/components/publicProfileInfo/styles.ts]: background-color should be taken from the theme

File: src/components/publicProfileSidebar/index.tsx (hunk 7 above)
Comment (AlexandraVoloshyna) [reply src/components/publicProfileSidebar/index.tsx:44]: This is not a comparison. There is nothing wrong here.

File: src/components/productCard/index.tsx (hunk 2 above)
Comment (eugeneiv1) [reply src/components/productCard/index.tsx:21]: the same TODO as for Product Name

File: src/components/productCard/index.tsx (hunk 1 above)
Comment (eugeneiv1) [reply src/components/productCard/index.tsx:11]: used to ignore unused variable until we fetch data from server

File: src/components/publicProfileInfo/index.tsx (hunk 3 above)
Comment (eugeneiv1) [reply src/components/publicProfileInfo/index.tsx]: fixed

File: src/components/publicProfileInfo/styles.ts (hunk 4 above)
Comment (eugeneiv1) [reply src/components/publicProfileInfo/styles.ts]: fixed

File: src/components/publicProfileInfo/types.ts (hunk 5 above)
Comment (eugeneiv1) [reply src/components/publicProfileInfo/types.ts]: fixed

File: src/components/publicProfileSidebar/styles.ts (hunk 9 above)
Comment (eugeneiv1) [reply src/components/publicProfileSidebar/styles.ts]: fixed

File: src/components/publicProfileInfo/styles.ts (hunk 4 above)
Comment (eugeneiv1) [reply src/components/publicProfileInfo/styles.ts]: fixed

File: src/components/publicProfileSidebar/styles.ts (hunk 8 above)
Comment (eugeneiv1) [reply src/components/publicProfileSidebar/styles.ts]: fixed

**Pull request merged**